bank.py

```diff
@@ -0,0 +1,166 @@
+import uuid
+from collections import UserDict
+from datetime import datetime
+from functools import wraps
+from typing import Dict, Tuple, Union
+
+from account import Account
+from history_tool import AddHistoryMixIn, History
+
+
+class AccountDoesNotExist(Exception):
+    pass 
+
+
+class AccountExist(Exception):
+    pass
+
+
+class InvalidOpeningBalance(Exception):
+    pass
+
+
+class InsufficientTransferBalance(Exception):
+    pass
+
+
+class DictAccount(UserDict):
+
+    def get(self, key: str):
+        """
+        Method return value if key exist 
+        If not method raise Exception.
+        """
+        try:
+            value = self[key]
+        except KeyError:
+            raise AccountDoesNotExist("Account about this id not exist")
+        return value  
+
+
+def payment_for_account(func):
+    """
+    Decorator checks if 200 operations have been performed on the account. 
+    If so, a fee is charged for each account.
+    """
+    @wraps(func)
+    def wrapped_function(*args, **kwargs):
+        bank = args[0]
+        if bank.uptime == 199:
+            bank.uptime = 0
+            for account in bank.accounts.values():
+                account.balance -= account.balance * bank.storage_commission
+        result = func(*args, **kwargs)
+        return result
+    return wrapped_function
+
+
+class Bank(AddHistoryMixIn):
+    
+    def __init__(self, name :str = 'PKO', storage_commission :float = 0.00000001,
+                 transfer_commission :float = 0.0000001) -> None:
+        super().__init__()
+        self.name = name
+        self.add_history(
+            message=f"Bank {self.name} has been founded.",
+            date=datetime.timestamp(datetime.utcnow())
+        )
+        self.transfer_commission = transfer_commission
+        self.storage_commission = storage_commission
+        self.accounts = DictAccount()
+        self.income = 0.00
+        self.uptime = 0
+
+    @payment_for_account
+    def open_account(self, name :str = '', balance :Union[int, float] = 0):
+        """
+        Method open new account.
+        """
+        self.uptime += 1
+        if balance < 0:
+            raise InvalidOpeningBalance("You can't open account with deficit.")
+        id = uuid.uuid4().hex
+        try:
+            account = self.accounts.get(id)
+        except AccountDoesNotExist:
+            pass
+        else:
+            if account.active != False:
+                raise AccountExist("Account about this id exists.")
+        self.accounts[id] = Account(
```

Comment:
i think, for the best experience, accounts should be an another class (as aggegator, see https://martinfowler.com/bliki/DDD_Aggregate.html) with methods like `add` `compute_commision` etc. For now you still have problem with change datasystem